cf_xarray/accessor.py

```diff
@@ -617,12 +617,12 @@ def check_results(names, k):
     successful = dict.fromkeys(key, False)
     for k in key:
         if "coords" not in skip and k in _AXIS_NAMES + _COORD_NAMES:
-            names = _get_axis_coord(obj, k)
```

Comment:
we should think about this one. when pulling out dataarrays we want the appropriate 'latitude' and 'longitude'...

Reply:
This is what I tried to break this (coordinate actually named latitude):
```python
from cf_xarray.datasets import popds
popds["ULAT"].attrs["standard_name"] = "latitude"
popds = popds.assign_coords(latitude=popds["ULAT"])
popds = popds.reset_coords("ULAT")
assert "latitude" not in popds.cf["UVEL"].coords
assert "ULAT" in popds.cf["UVEL"].coords
popds["UVEL"].attrs["coordinates"] = popds["UVEL"].attrs["coordinates"].replace("ULAT", "latitude")
assert "latitude" in popds.cf["UVEL"].coords
assert "ULAT" not in popds.cf["UVEL"].coords
```
Looks OK, and I can't think of other cases where this change wouldn't work. Any ideas?

Accidentally I found out that this raises an error:
```python
popds.cf.data_vars
```
Error is: Receive multiple variables for key 'latitude': {'ULAT', 'latitude'}. Expected only one. Please pass a list ['latitude'] instead to get all variables matching 'latitude'.

Not sure whether this is correct, as there's just one latitude `data_var` (but multiple `coords`), but it looks like we used to have the same behavior?

```
Coordinates:
- CF Axes: * X: ['nlon']
           * Y: ['nlat']
             Z, T: n/a

- CF Coordinates:   longitude: ['TLONG', 'ULONG']
                    latitude: ['TLAT', 'latitude']
                    vertical, time: n/a

- Cell Measures:   area, volume: n/a

- Standard Names:   latitude: ['latitude']

Data Variables:
- Cell Measures:   area, volume: n/a

- Standard Names:   latitude: ['ULAT']
                    sea_water_potential_temperature: ['TEMP']
                    sea_water_x_velocity: ['UVEL']
```

Reply:
> popds.cf.data_vars

Nice find! Definitely a bug. Let's open an issue for that.

The `ULAT` vs `latitude` issue is interesting. Fundamentally we need to decide whether cf-xarray treats xarray objects as "bags of variables" ignoring distinctions like data_vars and coords, or whether it fully follows xarray's data_vars vs coords. I'm leaning towards the former i.e. cf-xarray tries its best to attach variables mentioned in attributes ignoring any data_vars/coords distinction. I think we do this will ancillary variables for e.g. (?)


``` python
from cf_xarray.datasets import popds
popds["ULAT"].attrs["standard_name"] = "latitude"
popds = popds.assign_coords(latitude=popds["ULAT"])
popds = popds.reset_coords("ULAT")
assert "latitude" not in popds.cf["UVEL"].coords
assert "ULAT" in popds.cf["UVEL"].coords  
```

Reply:
Yes, I also think this is the best way to handle it (and I'm in favor of changing what's returned by properties such as `cf.coordinates` https://github.com/xarray-contrib/cf-xarray/issues/107#issuecomment-782908902)